server/index.js

```diff
@@ -23,103 +23,128 @@ const open = require('open');
 
-async function start(portPreferred) {
-    let port = process.env.ASSET_COMPUTE_DEV_PORT || '9000';
-    if (!isNaN(portPreferred)) {
-        port = portPreferred;
+class DevtoolServer {
+
+    async run (portPreferred) {
+        this.port = process.env.ASSET_COMPUTE_DEV_PORT || DEFAULT_PORT;
+        if (!isNaN(portPreferred)) {
+            this.port = portPreferred;
+        }
+    
+        const portRangeServer = {
+            port: this.port, // minimum port
+            stopPort: 9999 // maximum port
+        };
+        this.port = await findOpenPort(portRangeServer);
+
+        const randomString = getRandomString();
+    
+        
+    
+        // Get port from environment and store in Express.
+        app.set('port', this.port);
+        app.set('devToolToken', randomString);
+    
+        // Create HTTP server.
+        this.server = http.createServer(app);
+        this.httpTerminator = createHttpTerminator({ server: this.server }); // to be used in aio cli plugin to stop the server
+    
+        this.server.listen(this.port);
+        this.server.on('error', error => {
+            if (error.syscall !== 'listen') {
+                throw new Error(error);
+            }
+
+            // handle specific listen errors with friendly messages
+            const bind = typeof this.port === 'string'
+                ? 'Pipe ' + this.port
+                : 'Port ' + this.port;
+            switch (error.code) {
+            case 'EACCES':
+                console.error(bind + ' requires elevated privileges');
+                break;
+            case 'EADDRINUSE':
+                console.error(bind + ' is already in use');
+                break;
+            }
+            throw new Error(error);
+        });
+
+        this.server.on('listening', () => onListening(this.server, randomString));
+        this.server.on('close', () => {
+            console.log("Asset Compute Developer Tool Server Stopped");
+        });
+    }
+
+    async stop() {
+        return this.server.close();
     }
 
-    const portRangeServer = {
-        port: port, // minimum port
-        stopPort: 9999 // maximum port
-    };
-    port = await findOpenPort(portRangeServer);
+}
+
+// for backwards compatibility
+async function start(port) {
+    const devtool = new DevtoolServer();
+    await devtool.run(port);
+    return devtool;
+}
+
+/**
+ * Event listener for HTTP server "listening" event.
+ */
+async function onListening(server, randomString) {
+    const addr = server.address();
+    const bind = typeof addr === 'string'
+        ? 'pipe ' + addr
+        : 'port ' + addr.port;
+    debug('Listening on ' + bind);
+    if (process.env.ASSET_COMPUTE_DEV_TOOL_ENV !== 'development') {
+        const assetComputeDevToolUrl = `http://localhost:${addr.port}/?devToolToken=${randomString}`;
+        console.log('Asset Compute Developer Tool Server started on url', assetComputeDevToolUrl);
+        await open(assetComputeDevToolUrl);
```

Comment:
Switch L108 and L107: Only log the server has started once it has started.

Reply:
server has already started at this point. line 108 simply opens the url in the browser for the user. It makes more sense for the user to see this message displayed in the terminal before a browser pops open for them